### DISCUSSION TITLE
Why is our application calling route multiple times?

### DISCUSSION THREAD

**Discussion by kopax-polyconseil** [TanStack/query#2692] [Q&A]
We are working on [this application](https://github.com/pass-culture/pass-culture-app-native) which use `react-query@3.8.2`.

So far, we can see two react query hooks that are calling our endpoints too many times: `/me`, `/favorites`:

![image](https://user-images.githubusercontent.com/77674046/133999457-d41e9800-a525-4649-8794-98f2409a0da9.png)

![image](https://user-images.githubusercontent.com/77674046/133999446-66630aff-a285-4e3e-8f83-87708d5d8f58.png)

Our hooks looks like : 

```jsx
  const { isLoggedIn } = useAuthContext()

  return useQuery<PaginatedFavoritesResponse>(
    QueryKeys.FAVORITES,
    () => api.getnativev1mefavorites(),
    {
      enabled: isLoggedIn,
    }
  )
```

and 

```jsx

export function useUserProfileInfo(options = {}) {
  const { isLoggedIn } = useAuthContext()

  return useQuery<UserProfileResponse>(QueryKeys.USER_PROFILE, () => api.getnativev1me(), {
    enabled: isLoggedIn,
    ...options,
  })
}
```

We use a react query cache provider, we expected to see react-query handling the queries strategy, instead, we see multiple queries. Any clue what is going on wrong

**Comment by TkDodo** [comment]
please read the [Important Defaults](https://react-query.tanstack.com/guides/important-defaults) section in the docs. With the default staleTime of zero, react-query will do background updates to keep your data accurate.

**Reply by kopax-polyconseil** [reply]
Those requests seems to be called in a row, like if it was re-rendering and calling as much as it re-renders, is it expected with `react-query`? After reading the whole documentation and watching a few videos, it seemed to me that the ReactQueryCacheProvider should cache this and prevent such bug.

**Reply by TkDodo** [reply]
Impossible to say without some code. Possibly you are mounting instead of rerendering. Also, ReactQueryCacheProvider only exists on v2, not v3 🤔
